### DISCUSSION TITLE
Use local copy of font-awesome CSS instead of CloudFlare?

### DISCUSSION THREAD

**Discussion by hussam-qasem** [mojolicious/mojo#2062] [Q&A] [answered]
I noticed that [`mojo/debug.html.ep`](https://github.com/mojolicious/mojo/blob/addf3cafc6810fb9554cffc927c3797ba1b15fea/lib/Mojolicious/resources/templates/mojo/debug.html.ep#L16) loads:
```html
<link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.3/css/all.min.css" rel="stylesheet">
```
is it possible to make the CSS file available locally? or shall I change the template locally?

Kind regards.

**Comment by guest20** [comment, marked as answer]
The markdown sure did a number on that  url you pasted... 

Since this is a dev-only page it might make sense to just copy it into your own `templates/` directory and delete the lines you don't want. The css itself just adds some social media icons to the links in the footer. I'm sure you can live without that.

**Comment by kraih** [comment]
Yes, you can make your own replacement templates for 500 and 404. The built-in one is just a fallback.

**Comment by hussam-qasem** [comment]
Thank you @guest20 and @kraih.

I shall do it locally. I just wasn't sure why `font-awesome` wasn't part of Mojolicious like `Bootstrap`,` jQuery` and `highlight.js`.

The only reason I noticed was because my Internet dropped, and `debug` template took forever to load. Then I realized it was trying to fetch something (the CSS) from the Internet.

**Comment by kraih** [comment]
We could probably defer the loading of the stylesheet to make the framework deal with this situation a little more gracefully.
